lua/ocaml/init.lua

```diff
@@ -448,23 +432,37 @@ function M.decrease_verbosity()
   M.display_type(enclosings[index])
 end
 
-function M.start_session()
+--- Starts a type enclosing session and register keys
+---@param user_opts ocaml.config.OCamlConfig
+local function start_session(user_opts)
```

Comment:
As it is only used here and required to use the user_options, I don't think it is necessary to make it public. I had this comment to help for the review later and document the origin of the change.